### PR TITLE
Added upload service for uploading records and images.

### PR DESCRIPTION
**Pull request by tony-gaia**
Added photo count on record lists.

Renamed numerous images.

### CODE REVIEW

File: src/components/records-list/records-list.ts [hunk 1]
```diff
@@ -26,14 +26,25 @@ export class RecordsListComponent {
...
             case 'KLM-SAT Census':
-                return 'assets/imgs/koala_data_poop.png';
+                return 'assets/imgs/poop.png';
             case 'KLM-SAT Tree Sighting':
-                return 'assets/imgs/koala_data_tree.png';
+                return 'assets/imgs/tree.png';
+        }
+    }
+
+    public getCountIcon(record: ClientRecord): string {
```
Comment (GaryDaviesGaiaResources) [inline src/components/records-list/records-list.ts]: Method name doesnt match method behaviour.

File: src/shared/services/upload.service.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        );
+    }
+
+    public uploadRecordPhotos(record: Record, photoIds): Observable<boolean> {
+        return from(photoIds).pipe(
+            mergeMap((photoId: string) => this.storageService.getPhoto(photoId).pipe(
+                mergeMap((photo: ClientPhoto) => this.apiService.uploadRecordMediaBase64(record.id, photo.base64))),
+                (photoId: string, media: Media) => photoId
+                ),
+            mergeMap((photoId: string) => this.storageService.deletePhoto(photoId))
```
Comment (GaryDaviesGaiaResources) [inline src/shared/services/upload.service.ts]: Delete each on success

File: src/components/records-list/records-list.ts (hunk 1 above)
Comment (tony-gaia) [reply src/components/records-list/records-list.ts]: In this case these are the icons on the right side that indicate what's being counted, e.g. koalas for opportunistic obs and tree observations, and tree sighting count for census

File: src/shared/services/upload.service.ts (hunk 2 above)
Comment (tony-gaia) [reply src/shared/services/upload.service.ts]: It will only get to this point if there hasn't been an upload failure

**Pull request merged**